vxquery-cli/src/main/java/org/apache/vxquery/cli/VXQuery.java

```diff
@@ -120,200 +94,236 @@ public static void main(String[] args) throws Exception {
+    }
+
+    private void onQueryResultFetchSuccess(String xqFile, QueryResultRequest request, QueryResultResponse response) {
+        if (request.isShowMetrics()) {
+            String metrics = String.format("Elapsed Time:\t%d", response.getMetrics().getElapsedTime());
+            printField("Result Reading Metrics", metrics);
+        }
+
+        printField("Results", response.getResults());
+    }
+
+    private void onQueryFailure(String xqFile, ErrorResponse response) {
+        if (response == null) {
+            System.err.println(String.format("Unable to execute query in %s", xqFile));
+            return;
         }
+
+        System.err.println();
+        System.err.println("====================================================");
+        System.err.println("\t'" + xqFile + "' Errors");
+        System.err.println("====================================================");
+
+        Error error = response.getError();
+        String errorMsg = String.format("Code:\t %d\nMessage:\t %s", error.getCode(), error.getMessage());
+        printField(System.err, String.format("Unable to execute query in '%s'", xqFile), errorMsg);
     }
 
+
     /**
-     * Creates a Hyracks dataset, if not already existing with the job frame size, and 1 reader. Allocates a new buffer of size specified in the frame of Hyracks
-     * node. Creates new dataset reader with the current job ID and result set ID. Outputs the string in buffer for each frame.
+     * Submits a query to be executed by the REST API. Will call {@link #onQueryFailure(String, ErrorResponse)} if any
+     * error occurs when submitting the query. Else will call {@link #onQuerySubmitSuccess(String, QueryRequest,
+     * QueryResponse)} with the {@link QueryResponse}
      *
-     * @param spec
-     *            JobSpecification object, containing frame size. Current specified job.
-     * @param writer
-     *            Writer for output of job.
-     * @throws Exception
+     * @param xqFile  .xq file with the query to be executed
+     * @param request {@link QueryRequest} instance to be submitted to REST API
+     * @param cli     cli class instance
      */
-    private void runJob(JobSpecification spec, PrintWriter writer) throws Exception {
-        int nReaders = 1;
-        if (hds == null) {
-            hds = new HyracksDataset(hcc, spec.getFrameSize(), nReaders);
+    private static void sendQueryRequest(String xqFile, QueryRequest request, VXQuery cli) {
+        URI uri = null;
+        try {
+            uri = RestUtils.buildQueryURI(request, cli.restIpAddress, cli.restPort);
+        } catch (URISyntaxException e) {
+            System.err.println(String.format("Unable to build URI to call REST API for query: %s", request.getStatement()));
+            cli.onQueryFailure(xqFile, null);
         }
 
-        JobId jobId = hcc.startJob(spec, EnumSet.of(JobFlag.PROFILE_RUNTIME));
+        CloseableHttpClient httpClient = HttpClients.custom()
```

Comment:
This file should be formatted using AsterixDB standard. See the website for developer code formatting details.